frontend/packages/dev-console/src/components/pipelines/pipeline-form/PipelineSecretSection.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { Formik } from 'formik';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { Button } from '@patternfly/react-core';
+import { ExpandCollapse } from '@console/internal/components/utils';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { SecretModel } from '@console/internal/models';
+import { k8sCreate } from '@console/internal/module/k8s';
+import SecretForm from './SecretForm';
+import {
+  associateServiceAccountToSecret,
+  getSecretAnnotations,
+} from '../../../utils/pipeline-utils';
+import SecretsList from './SecretsList';
+import { advancedSectionValidationSchema } from './pipelineForm-validation-utils';
+import { SecretAnnotationId } from '../const';
+
+import './PipelineSecretSection.scss';
+
+const initialValues = {
+  secretName: '',
+  annotations: { key: SecretAnnotationId.Image, value: '' },
+  type: SecretType.dockerconfigjson,
+  formData: {},
+};
+
+type PipelineSecretSectionProps = {
+  namespace: string;
+  setsubmitDisabled: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const PipelineSecretSection: React.FC<PipelineSecretSectionProps> = ({
+  namespace,
+  setsubmitDisabled,
+}) => {
+  const [addSecret, setAddSecret] = React.useState(false);
```

Comment:
Perhaps just take this all the way up to the top-level form?

`useField('secretOpen')`

And use a valdiationSchema to make sure that is false.

ie, here do:
```
  const [{ value: secretOpen }] = useField<boolean>('secretOpen');
  const { setFieldValue } = useFormikContext<FormikValues>();
  const setSecretOpen = (isOpen: boolean) => setFieldValue('secretOpen', isOpen);
```
And add to the startPipelineSchema:
```
secretOpen: yup.boolean().equals([false]),
```

No threading, single point of reference for everything, form-level locking... I think this is just smoother.

Reply:
Moved it to the top level and added yup validation to handle the submit  button disabling.